docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md

```diff
@@ -154,3 +154,22 @@ unique keys to `failureDomainSpec`s as well as respecting a set `Machine.Spec.Fa
 instances.
 
 Please see the cluster and machine infrastructure provider specifications for more detail.
+
+# Apply the contract version label `cluster.x-k8s.io/<version>: version1,version2,version3` to your CRDs
+
+- Providers MUST set the `cluster.x-k8s.io/<version>` to all Custom Resource Definitions related to Cluster API starting with v1alpha3.
+- The label is a map from an API Version of Cluster API (contract) to your Custom Resource Definition versions.
+  - The value is a comma-delimited list of versions.
+  - Each value MUST point to an available version in your CRD Spec.
+- The label allows Cluster API controllers will perform automatic conversions for object references, Cluster API
+  picks the last available version in the list if multiple versions are found.
+- To apply the label to CRDs it's possible to use `commonLabels` in your `kustomize.yaml` file, usually in `config/crd`.
+
+In this example we show how to map a particular Cluster API contract version to your own CRD using Kustomize's `commonLabels` feature:
```

Comment:
Should we be suggesting or using `commonLabels` here? These will apply to much more than the CRD types. I'm wondering if we should try to find a more targeted way to set the labels for just the CRD types.

Reply:
The suggested change should only apply under the `crd` directory, so it only applies to the CRDs

Reply:
Ah, indeed. sorry for the noise.